### ISSUE TITLE
GCop 420 false positive

### ISSUE THREAD

**Issue by HamidTheGeek** [Geeksltd/GCop#298]
It shows me a warning for the following code

```c#
public SortedNews() { }

public SortedNews(NewsDto news, string[] titleAndDescription, string[] body)
{
    News = news;
    TitleAndDescription = titleAndDescription;
    Body = body;
}
```

**Comment by marjanjavid** [comment]
@HamidTheGeek  Would you please mention the warning message or warning code?

**Comment by HamidTheGeek** [comment]
Hi Marjan,
I have already mentioned the code in the title of this issue. It's related to having a parameterless public constructor with out body.

**Comment by marjanjavid** [comment]
Yes my bad. Thank u

**Comment by Foroughi** [comment]
@HamidTheGeek Hi Hamid, would you please catch me up why this is a false alarm? You have an empty constructor in your code! According to the rule logic, it SHOULD kick in in this scenario. am I missing something here ?

**Comment by HamidTheGeek** [comment]
In the following code:

```C#
class C1
{
    public int MyProperty { get; set; }
}

class C2
{
    public int MyProperty { get; set; }

    public ClassName() { }
}

class C3
{
    public int MyProperty { get; set; }

    public ClassName(int prop) => MyProperty = prop;
}

class C4
{
    public int MyProperty { get; set; }

    public ClassName() { }

    public ClassName(int prop) => MyProperty = prop;
}

class Test
{
    void M()
    {
        var o1 = new C1();
        var o2 = new C2();
        var o3 = new C3(); // ======== compile error
        var o4 = new C4();
    }
}
```

The C1 and C2 are the same so you can force the user to remove the empty constructor.
But the C3 and C4 are not as you can not create a new instance like this `new C3()` while you can do it for C4.

**Comment by Foroughi** [comment]
Fixed

**Comment by marjanjavid** [comment]
It is fixed on GCop2.6.1
